packages/@react-stately/color/src/useHexColorFieldState.ts

```diff
@@ -0,0 +1,133 @@
+    maxValue = defaultMaxValue,
+    step = 1,
+    value,
+    defaultValue,
+    onChange,
+    validationState
+  } = props;
+
+  const minColor = Color.parse(minValue);
+  const maxColor = Color.parse(maxValue);
+  const minColorInt = minColor.toHexInt();
+  const maxColorInt = maxColor.toHexInt();
+
+  const clampColor = useCallback((value: ColorInput) => {
+    try {
+      const color = Color.parse(value);
+      const colorInt = color.toHexInt();
+      if (colorInt < minColorInt) { return minColor; }
+      if (colorInt > maxColorInt) { return maxColor; }
+      return color;
+    } catch (err) {
+      return undefined;
+    }
+  }, [minColorInt, maxColorInt]);
+
+  const initialValue = clampColor(value);
+  const initialDefaultValue = clampColor(defaultValue);
+  const [colorValue, setColorValue] = useControlledState<Color>(initialValue, initialDefaultValue, onChange);
+
+  const initialInputValue = (value || defaultValue) && colorValue ? colorValue.toString('hex') : '';
+  const [inputValue, setInputValue] = useState(initialInputValue);
+
+  const increment = () => {
+    setColorValue((previousValue: Color) => {
+      const colorInt = previousValue ? previousValue.toHexInt() : minColorInt;
+      const newValue = `#${Math.min(colorInt + step, maxColorInt).toString(16).padStart(6, '0')}`;
+      const newColor = new Color(newValue);
+      setInputValue(newColor.toString('hex'));
+      return newColor;
+    });
+  };
+
+  const incrementToMax = useCallback(() => {
+    if (maxValue) {
+      setColorValue(maxColor);
+      setInputValue(maxColor.toString('hex'));
+    }
+  }, [maxValue, maxColor, setColorValue, setInputValue]);
+
+  const decrement = () => {
+    setColorValue((previousValue: Color) => {
+      if (!previousValue) { return minColor; }
+      const colorInt = previousValue.toHexInt();
+      const newValue = `#${Math.max(colorInt - step, minColorInt).toString(16).padStart(6, '0')}`;
+      const newColor = new Color(newValue);
+      setInputValue(newColor.toString('hex'));
+      return newColor;
+    });
+  };
+
+  const decrementToMin = useCallback(() => {
+    if (minValue) {
+      setColorValue(minColor);
+      setInputValue(minColor.toString('hex'));
+    }
+  }, [minValue, minColor, setColorValue, setInputValue]);
+
+  const setFieldInputValue = (value: string) => {
+    setInputValue(value);
+    value = value.trim();
+    if (!value.length) { return; }
+    if (!value.startsWith('#')) {
+      value = `#${value}`;
+    }
+    try {
+      const color = clampColor(value);
+      if (color) {
+        setColorValue(color);
+      }
+    } catch (err) {
+      // ignore
+    }
+  };
+
+  const commitInputValue = () => {
+    setFieldInputValue(colorValue ? colorValue.toString('hex') : '');
+  };
+
+  return {
+    colorValue,
+    inputValue,
+    setInputValue: setFieldInputValue,
+    commitInputValue,
+    increment,
+    incrementToMax,
+    decrement,
+    decrementToMin,
```

Comment:
Increment and Decrement are not particularly useful one integer value at a time, I wonder if `incrementPage` and `decrementPage` could be added to increment and decrement, snapping to the closest Web Safe or named color value.

Reply:
Don't worry about this one in this PR